browser/modules/DirectoryLinksProvider.jsm

```diff
@@ -720,20 +720,19 @@ let DirectoryLinksProvider = {
         // Only add data for non-empty tiles
         if (site) {
           // Remember which tiles data triggered the action
-          let link = site;
           let tilesIndex = tiles.length;
           if (triggeringSiteIndex == pos) {
             actionIndex = tilesIndex;
           }
 
           // Make the payload in a way so keys can be excluded when stringified
-          let id = link.directoryId;
+          let id = site.directoryId;
           tiles.push({
             id: id || site.enhancedId,
-            pin: link.pinState ? 1 : undefined,
+            pin: site.pinState ? 1 : undefined,
```

Comment:
I also didn't write this original piece of code but that's a great question. I don't know the nitty gritty differences between the two but I think it would be good for us to make some coding guidelines for which to use when...